src/mongocxx/collection.cpp

```diff
@@ -197,20 +196,20 @@ void* collection::implementation() const {
     return _impl->collection_t;
 }
 
-stdx::optional<result::insert_one> collection::insert_one(bsoncxx::document::view document,
+stdx::optional<result::insert_one> collection::insert_one(bsoncxx::document::view_or_value document,
                                                           const options::insert& options) {
     class bulk_write bulk_op(false);
     bsoncxx::document::element oid{};
 
     bsoncxx::builder::stream::document new_document;
-    if (!document["_id"]) {
+    if (!document.view()["_id"]) {
```

Comment:
I don't think so.  The view_or_value class should stay as vanilla as possible, and operator [] only makes sense when we're using it with bson documents.  Eventually I think we'll want to use view_or_value to handle string passing, and maybe some other things.  I suppose we could add an operator[] that just called operator[] on the view, but it still feels a bit clunky to me when I consider view_or_value as an independent class.